build.gradle

```diff
@@ -362,6 +405,17 @@ dependencies {
     implementation "com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}"
 
     compileOnly "org.opensearch:opensearch:${opensearch_version}"
+
+    //integration test framework:
+    integrationTestImplementation('com.carrotsearch.randomizedtesting:randomizedtesting-runner:2.7.1') {
+        exclude(group: 'junit', module: 'junit')
+    }
+    integrationTestImplementation 'junit:junit:4.13.2'
+    integrationTestImplementation "org.opensearch.plugin:reindex-client:${opensearch_version}"
+    integrationTestImplementation "org.opensearch.plugin:percolator-client:${opensearch_version}"
+    integrationTestImplementation 'commons-io:commons-io:2.7'
+    integrationTestImplementation 'org.apache.logging.log4j:log4j-core:2.17.1'
+    integrationTestImplementation 'org.hamcrest:hamcrest:2.1'
```

Comment:
Yes, we can but it will not match version configured in the OpenSearch project. Maybe it would be better to add:

`integrationTestImplementation "org.opensearch.test:framework:${opensearch_version}",` since org.opensearch.test:framework is also automatically added as a test dependency when the opensearch.opensearchplugin plugin is used.
Thanks to that we will be able to get rid of:

    integrationTestImplementation('com.carrotsearch.randomizedtesting:randomizedtesting-runner:2.7.1') {
        exclude(group: 'junit', module: 'junit')
    }
    integrationTestImplementation 'junit:junit:4.13.2'
    integrationTestImplementation "org.hamcrest:hamcrest:2.2"

because org.opensearch.test:framework provides these dependencies. What do you think?